powerflex_data/data_collection.py

```diff
@@ -0,0 +1,124 @@
+import s3fs
+import boto3
+import pandas as pd
+import requests
+import json
+import time
+import datetime
+import sys
+from io import StringIO
+import boto3
+
+# url for interval data
+URL_MEASUREMENT = "https://slac.powerflex.com:9443/get_measurement_data"
+
+# urls for session data from two different sites
+URL_ARCH1 = "https://archive.powerflex.com/get_csh/0032/01"
+URL_ARCH2 = "https://archive.powerflex.com/get_csh/0032/02"
+
+URL_LOGIN = "https://slac.powerflex.com:9443/login"
+URL_ARCHIVE = "https://archive.powerflex.com/login"
+
+def api_login(username, password):
+    data_login = json.dumps({"username": username, "password": password})
+    headers = {'cache-control': 'no-cache', 'content-type': 'application/json'}
+
+    resp_login = requests.post(URL_LOGIN, headers=headers, data=data_login)
+    resp_archive = requests.post(URL_ARCHIVE, headers=headers, data=data_login)
+
+    token_meas = json.loads(resp_login.text)["access_token"]
+    auth_token_meas = "Bearer " + token_meas
+    headers_meas = {"cache-control": "no-cache", "content-type":
+                    "application/json", "Authorization": auth_token_meas}
+
+    token_arch = json.loads(resp_archive.text)["access_token"]
+    auth_token_arch = "Bearer " + token_arch
+    headers = {"cache-control": "no-cache", "content-type": "application/json",
+               "Authorization": auth_token_arch}
+    return (headers_meas, headers)
```

Comment:
Two login requests in one function that do the exact same thing. DRY this up and make it generic.

Reply:
Done.